// express/src/controllers/productsController.js

var productDAO = require('../models/productDAO');
var ProductDTO = require('../models/productDTO');

function ProductController(){}

ProductController.prototype.indexAction= function(request, response){

	function select(err, products){
		// console.log(products);
		response.render('products',{products:products});
	}
	productDAO.findAll(select);
};

ProductController.prototype.showAction= function(request, response, next){
	var id=request.params.id;
	function select(err, product){
		
		response.render('product',{product:product});
	}
	productDAO.findOne(select, id);
};

ProductController.prototype.insertAction= function(request, response){
	productDTO = new ProductDTO();
	var post=request.body;

	productDTO.setName(post.name);
	productDTO.setDesc(post.desc);
	productDTO.setPrice(post.price);
	productDTO.setImage(post.image);
	
	productDAO.insert(productDTO);
	response.redirect('/products');
};

ProductController.prototype.deleteAction= function(request, response){
	var id=request.params.id;
		
	productDAO.delete(id);
	response.redirect('/products');
};

module.exports = new ProductController();
